device/arista/x86_64-arista_7050_qx32s/plugins/led_control.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+#
+# led_control.py
+# 
+# Platform-specific LED control functionality for SONiC
+#
+
+try:
+    from sonic_led.led_control_base import LedControlBase
+except ImportError, e:
+    raise ImportError (str(e) + " - required module not found")
+
+
+class LedControl(LedControlBase):
+    """Platform specific LED control class"""
+
+    SONIC_PORT_NAME_PREFIX = "Ethernet"
+
+    LED_SYSFS_PATH_BREAKOUT_CAPABLE = "/sys/class/leds/qsfp{0}_{1}/brightness"
+    LED_SYSFS_PATH_NO_BREAKOUT = "/sys/class/leds/qsfp{0}/brightness"
+
+    QSFP_BREAKOUT_START_IDX = 5
+    QSFP_BREAKOUT_END_IDX = 28
+    QSFP_NO_BREAKOUT_START_IDX = 29
+    QSFP_NO_BREAKOUT_END_IDX = 36
+
+    LED_COLOR_OFF = 0
+    LED_COLOR_GREEN = 1
+    LED_COLOR_YELLOW = 2
+
+    # Helper method to map SONiC port name to Arista QSFP index
+    def _port_name_to_qsfp_index(self, port_name):
+        # Strip "Ethernet" off port name
+        if not port_name.startswith(self.SONIC_PORT_NAME_PREFIX):
+            return -1
+
+        sonic_port_num = int(port_name[len(self.SONIC_PORT_NAME_PREFIX):])
+
+        # SONiC port nums are 0-based and increment by 4
+        # Arista QSFP indices are 1-based and increment by 1
+        return ((sonic_port_num/4) + 5)
+
+    # Concrete implementation of port_link_state_change() method
+    def port_link_state_change(self, port, state):
+        qsfp_index = self._port_name_to_qsfp_index(port)
+        
+        # Ignore invalid QSFP indices
+        if qsfp_index <= 0:
+            return
+
+        # QSFP indices 1-24 are breakout-capable and have four LEDs, and each LED indicate one lane.
+        # whereas indices 25-32 are not breakout-capable, and only have one
```

Comment:
Comment doesn't match indices as defined above. Comment should reference the constant names instead of actual values in both this plugin and the plugin for the x86_64-arista_7050_qx32 platform to avoid this in the future.